### PR TITLE
Tegra fixes from downstream

### PR DESCRIPTION
**Pull request by vwadekar**
Bug fixes from the downstream repo

### CODE REVIEW

File: plat/nvidia/tegra/common/tegra_bl31_setup.c [hunk 1]
```diff
@@ -379,13 +379,12 @@ void bl31_plat_arch_setup(void)
...
 {
-	uint64_t end = base + size_in_bytes - 1;
+	uint64_t end = base + size_in_bytes;
 
 	/*
 	 * Check if the NS DRAM address is valid
 	 */
-	if ((base < TEGRA_DRAM_BASE) || (end > TEGRA_DRAM_END) ||
-	    (base >= end)) {
+	if ((base < TEGRA_DRAM_BASE) || (end > TEGRA_DRAM_END)) {
```
Comment (davidcunado-arm) [inline plat/nvidia/tegra/common/tegra_bl31_setup.c]: Just to check... 'end' points to the address one after size_in_bytes starting from base; assume that TEGRA_DRAM_END also points to the address one after the last available / valid address?

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by davidcunado-arm** [comment]
@vwadekar 
Looks like a missing definition for t132 platform:

/tegra/t132/debug/bl31/tegra_bl31_setup.d -MT build/tegra/t132/debug/bl31/tegra_bl31_setup.o -MP -c plat/nvidia/tegra/common/tegra_bl31_setup.c -o build/tegra/t132/debug/bl31/tegra_bl31_setup.o
plat/nvidia/tegra/common/tegra_bl31_setup.c: In function 'bl31_plat_arch_setup':
plat/nvidia/tegra/common/tegra_bl31_setup.c:358:14: error: 'TEGRA_TMRUS_SIZE' undeclared (first use in this function)
    (uint64_t)TEGRA_TMRUS_SIZE,
              ^
plat/nvidia/tegra/common/tegra_bl31_setup.c:358:14: note: each undeclared identifier is reported only once for each function it appears in
make: *** [build/tegra/t132/debug/bl31/tegra_bl31_setup.o] Error 1

File: plat/nvidia/tegra/common/tegra_bl31_setup.c (hunk 1 above)
Comment (vwadekar) [inline plat/nvidia/tegra/common/tegra_bl31_setup.c]: Correct.

**Comment by vwadekar** [comment]
@davidcunado-arm 

The change for T132 somehow didnt make it into the upstream repo. Fixed it now.

File: include/lib/stdlib/string.h [hunk 2]
```diff
@@ -55,6 +55,7 @@ void	*memcpy(void * __restrict, const void * __restrict, size_t);
 void	*memcpy16(void * __restrict, const void * __restrict, size_t);
 void	*memmove(void *, const void *, size_t);
 void	*memset(void *, int, size_t);
+void	 zeromem16(void *, unsigned int);
```
Comment (davidcunado-arm) [inline include/lib/stdlib/string.h]: The zeromem16 function is not one that we want to export - we currently using it internally only due to its focused alignment requirements.
Options are

- Use the new, more general purpose zeromem functions in utils.h
- Or if you wish to keep using the zeromem16, keep the extern declaration where it currently is

Comment (vwadekar) [inline include/lib/stdlib/string.h]: > Use the new, more general purpose zeromem functions in utils.h

@davidcunado-arm, where can I find this header? Dont see it in the tree.

**Comment by davidcunado-arm** [comment]
@vwadekar 
It's here -> https://github.com/ARM-software/arm-trusted-firmware/blob/master/include/lib/utils.h

**Comment by vwadekar** [comment]
@davidcunado-arm 

Thanks. I will update the PR.

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by vwadekar** [comment]
Thanks David.

**Pull request merged**